### PR TITLE
support metadata files as CLI arg supplier

### PR DESCRIPTION
**Pull request by anthonywu**
Oct 19 - this is a preview of a big change that I think we'll discuss for several days. It probably should be in the (n+1)th release. I expect we'll go back and forth a few times here because the decisions here influence how other subsequent features are done.

-----

# Change

1. Add a new CLI option to consume the run configs (but not the model configs) from `.json` files. This allows for very long commands with many path args to be easily re-usable for subsequent runs. You can imagine this being useful for anyone experimenting with _changing one thing_ and seeing how that single variable affects the outcome.
1. Promote the utility of the existing metadata upgrade files to serve as the input for run configs. The configs are simple enough to be handcrafted, but the metadata output is the automatically compatible one we will offer.
1. There is some metadata schema change here, but it was not consumed anywhere by mflux so I would not consider this compatibility breaking for SemVar purposes.
1. See more inline comments for todos and design concerns.

# UX Demo

What the change allows are commands like:

1. `mflux-generate --model dev --metadata --config-from-metadata <file>` - re-generate from yours or other's prior run configs. You can use these as references for checking into team repos, or to use as the control config in a test suite where the before/after of a image gen change can from the same config file defined in a repo.
1. `mflux-generate --model schnell --metadata --config-from-metadata <file from dev run>` - re-do the image gen, but using a diff model than the one that saved the config
1. `mflux-generate --model dev --metadata --config-from-metadata <file> --guidance 5.0` - re-genenerate, but change 1 variable
1. `mflux-generate --model dev --metadata --config-from-metadata <file> --quantize 4` - re-generate, but with the lighter quantized model
1. put it in a for-loop somewhere (shell or python), and each iteration you change one thing. This is where my output namer proposal in #80 is relevant, i.e. I don't want to rename things as I iterate through large batches, I would opt into a smart auto-namer in that scenario.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,6 +152,76 @@ mflux-generate --model dev --prompt "Luxury food photograph" --steps 25 --seed 2
 
 - **`--controlnet-save-canny`** (optional, bool, default: False): If set, saves the Canny edge detection reference image used by ControlNet.
 
+- **`--config-from-metadata`** or **`-C`** (optional, `str`): [EXPERIMENTAL] Path to a prior file saved via `--metadata`, or a compatible handcrafted config file adhering to the expected args schema.
```
Comment (anthonywu) [inline README.md:155]: Discussion in #80 reminded me that we also export exif data to image files, so that opens out the possibility that you can pass in an image path here to, and if you pass in an image path, that can act as `--init-image` or `--controlnet-image-path` implicitly!

File: README.md [hunk 2]
```diff
@@ -152,6 +152,76 @@ mflux-generate --model dev --prompt "Luxury food photograph" --steps 25 --seed 2
...
+    },
+    "lora_scales": {
+      "type": ["array", "null"],
+      "items": {
+        "type": "number"
+      }
+    },
+    "prompt": {
+      "type": ["string", "null"]
+    }
```
Comment (anthonywu) [inline README.md:198]: for now just the basic image gen + lora args, withholding the controlnet implementation for now, because there are plans to dedupe/consolidate the two classes together so that would cause change in this PR

File: README.md [hunk 3]
```diff
@@ -152,6 +152,76 @@ mflux-generate --model dev --prompt "Luxury food photograph" --steps 25 --seed 2
...
+  "guidance": 3.0,
+  "quantize": 4,
+  "lora_paths": [
+    "/some/path1/to/subject.safetensors",
+    "/some/path2/to/style.safetensors"
+  ],
+  "lora_scales": [
+    0.8,
+    0.4
+  ],
```
Comment (anthonywu) [inline README.md:219]: these config files really help when this list grows!

File: src/mflux/config/config.py [hunk 4]
```diff
@@ -16,7 +16,7 @@ def __init__(
         width: int = 1024,
         height: int = 1024,
         guidance: float = 4.0,
-        init_image: Path | None = None,
+        init_image_path: Path | None = None,
```
Comment (anthonywu) [inline src/mflux/config/config.py:19]: all these `init_image_path` changes here can be manually cherry-picked into the next release, if you consider this entire change to be too much for a `0.4.0`

File: src/mflux/config/runtime_config.py [hunk 5]
```diff
@@ -49,9 +49,17 @@ def precision(self) -> mx.Dtype:
...
         return self.model_config.num_train_steps
 
+    @property
+    def init_image_path(self) -> int:
+        return self.config.init_image_path
+
+    @property
+    def init_image_strength(self) -> int:
+        return self.config.init_image_strength
+
```
Comment (anthonywu) [inline src/mflux/config/runtime_config.py:59]: I think we have some anti-patterns in the relation between "runtime config" and "model config", there's a lot of proxying things through nested config objects. I have some refactoring ideas here but for now I want to keep the config structure as-is so I don't bloat the change and add unrelated changes to the actual intent.

File: src/mflux/post_processing/generated_image.py [hunk 6]
```diff
@@ -22,8 +22,10 @@ def __init__(
         generation_time: float,
         lora_paths: list[str],
         lora_scales: list[float],
-        controlnet_image_path: str | None = None,
+        controlnet_image_path: str | pathlib.Path | None = None,
```
Comment (anthonywu) [inline src/mflux/post_processing/generated_image.py:25]: one of the nice things about `argparser` is you can `type=Path` and it's wrapped and validated for us at entry, this allows for that future update to `type=Path`

File: src/mflux/post_processing/generated_image.py [hunk 7]
```diff
@@ -36,29 +38,38 @@ def __init__(
...
+            "quantize": self.quantization,
+            "generation_time_seconds": round(self.generation_time, 2),
+            "lora_paths": [str(p) for p in self.lora_paths] if self.lora_paths else None,
+            "lora_scales": [round(scale, 2) for scale in self.lora_scales] if self.lora_scales else None,
+            "init_image_path": str(self.init_image_path) if self.init_image_path else None,
+            "init_image_strength": self.init_image_strength if self.init_image_path else None,
+            "controlnet_image_path": str(self.controlnet_image_path) if self.controlnet_image_path else None,
+            "controlnet_strength": round(self.controlnet_strength, 2) if self.controlnet_strength else None,
             "prompt": self.prompt,
-            "controlnet_image": "None" if self.controlnet_image is None else self.controlnet_image,
```
Comment (anthonywu) [inline src/mflux/post_processing/generated_image.py:60]: I noticed these were string `"None"` and therefore were `json` strings, was this to accommodate tools like `exif` data writing, or just oversight? I made them into proper `json` nulls for easier reading back.

If string casting is required for some destination, we can always re-cast it at that time.

File: src/mflux/post_processing/generated_image.py [hunk 8]
```diff
@@ -36,29 +38,38 @@ def __init__(
...
-            "lora_paths": ", ".join(self.lora_paths) if self.lora_paths else "None",
-            "lora_scales": ", ".join([f"{scale:.2f}" for scale in self.lora_scales]) if self.lora_scales else "None",
+            "seed": self.seed,
+            "steps": self.steps,
+            "guidance": self.guidance if ModelConfig.FLUX1_DEV else None,  # only the dev model supports guidance
+            "precision": str(self.precision),
+            "quantize": self.quantization,
+            "generation_time_seconds": round(self.generation_time, 2),
+            "lora_paths": [str(p) for p in self.lora_paths] if self.lora_paths else None,
+            "lora_scales": [round(scale, 2) for scale in self.lora_scales] if self.lora_scales else None,
```
Comment (anthonywu) [inline src/mflux/post_processing/generated_image.py:67]: made these into json arrays instead of csv strings, for easier parsing back

File: src/mflux/post_processing/generated_image.py [hunk 9]
```diff
@@ -36,29 +38,38 @@ def __init__(
...
-            "precision": f"{self.precision}",
-            "quantization": "None" if self.quantization is None else f"{self.quantization} bit",
-            "generation_time": f"{self.generation_time:.2f} seconds",
-            "lora_paths": ", ".join(self.lora_paths) if self.lora_paths else "None",
-            "lora_scales": ", ".join([f"{scale:.2f}" for scale in self.lora_scales]) if self.lora_scales else "None",
+            "seed": self.seed,
+            "steps": self.steps,
+            "guidance": self.guidance if ModelConfig.FLUX1_DEV else None,  # only the dev model supports guidance
+            "precision": str(self.precision),
+            "quantize": self.quantization,
```
Comment (anthonywu) [inline src/mflux/post_processing/generated_image.py:64]: made this arg consistent with cli arg name and we should all understand it's unit "bits"

File: src/mflux/post_processing/generated_image.py [hunk 10]
```diff
@@ -36,29 +38,38 @@ def __init__(
...
-            "steps": str(self.steps),
-            "guidance": "None" if self.model_config == ModelConfig.FLUX1_SCHNELL else str(self.guidance),
-            "precision": f"{self.precision}",
-            "quantization": "None" if self.quantization is None else f"{self.quantization} bit",
-            "generation_time": f"{self.generation_time:.2f} seconds",
-            "lora_paths": ", ".join(self.lora_paths) if self.lora_paths else "None",
-            "lora_scales": ", ".join([f"{scale:.2f}" for scale in self.lora_scales]) if self.lora_scales else "None",
+            "seed": self.seed,
+            "steps": self.steps,
+            "guidance": self.guidance if ModelConfig.FLUX1_DEV else None,  # only the dev model supports guidance
```
Comment (anthonywu) [inline src/mflux/post_processing/generated_image.py:62]: I noticed the prior implementation and accepted the logic as is, but want to double check this is true: `only the dev model supports guidance`

File: src/mflux/post_processing/generated_image.py [hunk 11]
```diff
@@ -36,29 +38,38 @@ def __init__(
...
         ImageUtil.save_image(self.image, path, self._get_metadata(), export_json_metadata)
 
     def _get_metadata(self) -> dict:
+        """Generate metadata for reference as well as input data for
+        command line --config-from-metadata arg in future generations.
+        """
         return {
+            # mflux_version is used by future metadata readers
+            # to determine supportability of metadata-derived workflows
             "mflux_version": str(GeneratedImage.get_version()),
```
Comment (anthonywu) [inline src/mflux/post_processing/generated_image.py:58]: we have choices:

1. don't support metadata files before 0.4.0 as config files
2. add some UX help to accept old metadata files

File: src/mflux/ui/cli/parsers.py [hunk 12]
```diff
@@ -7,33 +8,46 @@
 # fmt: off
 class CommandLineParser(argparse.ArgumentParser):
 
+    def __init__(self, *pargs, **kwargs):
+        super().__init__(*pargs, **kwargs)
+        self.supports_metadata_config = False
+        self.supports_image_generation = False
+        self.supports_controlnet = False
+        self.supports_image_to_image = False
+        self.supports_lora = False
```
Comment (anthonywu) [inline src/mflux/ui/cli/parsers.py:17]: this helps with mostly not doing `hasattr` checks

**Review by anthonywu — COMMENTED** [review]
for early discussion - several iterations expected to come from review

File: src/mflux/ui/cli/parsers.py [hunk 13]
```diff
@@ -7,33 +9,52 @@
...
+
+    def add_model_arguments(self, path_type: t.Literal["load", "save"] = "load") -> None:
+
+        if path_type == "load":
+            group = self.add_mutually_exclusive_group(required=True)
+            group.add_argument("--model", "-m", type=str, choices=ui_defaults.MODEL_CHOICES, help=f"The model to use ({' or '.join(ui_defaults.MODEL_CHOICES)}).")
+            group.add_argument("--path", type=str, default=None, help="Local path for loading a model from disk")
+        else:
+            self.add_argument("--model", "-m", type=str, required=True, choices=ui_defaults.MODEL_CHOICES, help=f"The model to use ({' or '.join(ui_defaults.MODEL_CHOICES)}).")
+            self.add_argument("--path", type=str, required=True, help="Local path for saving a model.")
```
Comment (anthonywu) [inline src/mflux/ui/cli/parsers.py:28]: @filipstrand this diff does 3 things:

1. important - fixes a real bug introduced in #75 - so you'll need to accept this PR or cherry-pick these changes before release. I'll try to cherry-pick into a smaller fix-it PR, as time allows in the next few days
1. re-differentiate the two diff meanings of `--path`, one is for loading, one is the target of `mflux-save`. In save context it's required, in the load context it's not required.
2. take advantage of mutually exclusive helper in `argparser`

**Comment by anthonywu** [comment]
@filipstrand in this PR I made major changes to the argparser expectations via adding the new config from file feature, so I added a test suite for all the various entry points. If you find any problems in `main`, we can:

1. port some of these fixes and tests to main, and we stay in PR to discuss the config loader feature as a 0.5.0 release
2. you find this PR acceptable as part of 0.4.0 release scope and we go out with the new tests

**Review by anthonywu — COMMENTED** [review]
test results at this checkpoint:

```
================================================================================== test session starts ==================================================================================
platform darwin -- Python 3.11.9, pytest-8.3.3, pluggy-1.5.0 -- /Users/anthonywu/Desktop/workspace/mflux-ops/.venv/bin/python3
cachedir: .pytest_cache
rootdir: /Users/anthonywu/workspace/mflux-ops
configfile: pyproject.toml
testpaths: tests
plugins: anyio-4.6.0, timer-1.0.0
collected 17 items
run-last-failure: rerun previous 2 failures first

tests/test_generate_image_controlnet.py::TestImageGeneratorControlnet::test_image_generation_dev_controlnet PASSED                                                                [  5%]
tests/test_generate_image_controlnet.py::TestImageGeneratorControlnet::test_image_generation_dev_lora_controlnet PASSED                                                           [ 11%]
tests/test_cli_argparser.py::test_model_path_requires_model_arg PASSED                                                                                                            [ 17%]
tests/test_cli_argparser.py::test_prompt_arg PASSED                                                                                                                               [ 23%]
tests/test_cli_argparser.py::test_guidance_arg PASSED                                                                                                                             [ 29%]
tests/test_cli_argparser.py::test_quantize_arg PASSED                                                                                                                             [ 35%]
tests/test_cli_argparser.py::test_seed_arg PASSED                                                                                                                                 [ 41%]
tests/test_cli_argparser.py::test_steps_arg PASSED                                                                                                                                [ 47%]
tests/test_cli_argparser.py::test_lora_args PASSED                                                                                                                                [ 52%]
tests/test_cli_argparser.py::test_image_to_image_args PASSED                                                                                                                      [ 58%]
tests/test_cli_argparser.py::test_controlnet_args PASSED                                                                                                                          [ 64%]
tests/test_cli_argparser.py::test_save_args PASSED                                                                                                                                [ 70%]
tests/test_generate_image.py::TestImageGenerator::test_image_generation_schnell PASSED                                                                                            [ 76%]
tests/test_generate_image.py::TestImageGenerator::test_image_generation_dev PASSED                                                                                                [ 82%]
tests/test_generate_image.py::TestImageGenerator::test_image_generation_dev_lora PASSED                                                                                           [ 88%]
tests/test_generate_image.py::TestImageGenerator::test_image_generation_dev_image_to_image PASSED                                                                                 [ 94%]
tests/test_generate_image_controlnet.py::TestImageGeneratorControlnet::test_image_generation_schnell_controlnet PASSED                                                            [100%]

===================================================================================== pytest-timer ======================================================================================
[success] 22.17% tests/test_generate_image_controlnet.py::TestImageGeneratorControlnet::test_image_generation_dev_lora_controlnet: 152.2725s
[success] 22.05% tests/test_generate_image_controlnet.py::TestImageGeneratorControlnet::test_image_generation_dev_controlnet: 151.4941s
[success] 16.29% tests/test_generate_image.py::TestImageGenerator::test_image_generation_dev: 111.8881s
[success] 16.09% tests/test_generate_image.py::TestImageGenerator::test_image_generation_dev_lora: 110.5325s
[success] 15.10% tests/test_generate_image.py::TestImageGenerator::test_image_generation_dev_image_to_image: 103.7386s
[success] 5.61% tests/test_generate_image_controlnet.py::TestImageGeneratorControlnet::test_image_generation_schnell_controlnet: 38.5621s
[success] 2.69% tests/test_generate_image.py::TestImageGenerator::test_image_generation_schnell: 18.5004s
[success] 0.00% tests/test_cli_argparser.py::test_model_path_requires_model_arg: 0.0006s
[success] 0.00% tests/test_cli_argparser.py::test_guidance_arg: 0.0004s
[success] 0.00% tests/test_cli_argparser.py::test_lora_args: 0.0004s
[success] 0.00% tests/test_cli_argparser.py::test_prompt_arg: 0.0004s
[success] 0.00% tests/test_cli_argparser.py::test_image_to_image_args: 0.0004s
[success] 0.00% tests/test_cli_argparser.py::test_quantize_arg: 0.0003s
[success] 0.00% tests/test_cli_argparser.py::test_seed_arg: 0.0003s
[success] 0.00% tests/test_cli_argparser.py::test_steps_arg: 0.0003s
[success] 0.00% tests/test_cli_argparser.py::test_controlnet_args: 0.0003s
[success] 0.00% tests/test_cli_argparser.py::test_save_args: 0.0002s
============================================================================ 17 passed in 688.04s (0:11:28) =============================================================================
```

File: README.md (hunk 1 above)
Comment (filipstrand) [reply README.md:155]: Nice idea, I think that would work!

File: src/mflux/config/config.py (hunk 4 above)
Comment (filipstrand) [reply src/mflux/config/config.py]: I think this feature is good to include in the `0.4.0` so we don't need to cherrypick.

File: src/mflux/config/runtime_config.py (hunk 5 above)
Comment (filipstrand) [reply src/mflux/config/runtime_config.py:49]: Sounds good! I agree with you that it is not the optimal way of structuring things... More generally, it is also a bit unclear what exactly should go in to the config. For example, now the seed is not included there, but one could argue that it is a type of config etc.

File: src/mflux/post_processing/generated_image.py (hunk 11 above)
Comment (filipstrand) [reply src/mflux/post_processing/generated_image.py:58]: Option 1. is fine I think

File: README.md [hunk 14]
```diff
@@ -152,6 +152,76 @@ mflux-generate --model dev --prompt "Luxury food photograph" --steps 25 --seed 2
...
+    }
+  }
+}
+```
+
+#### Example
+
+```json
+{
+  "model": "dev",
```
Comment (filipstrand) [inline README.md:207]: since the metadata file specifies the model also, should it be possible to launch a run purely from a config? I.e. 

`mflux-generate --config-from-metadata image.json`

Right now, it complains that we need to specify the model

`generate.py: error: the following arguments are required: --model/-m`

File: README.md [hunk 15]
```diff
@@ -152,6 +152,76 @@ mflux-generate --model dev --prompt "Luxury food photograph" --steps 25 --seed 2
...
+      }
+    },
+    "prompt": {
+      "type": ["string", "null"]
+    }
+  }
+}
+```
+
+#### Example
```
Comment (filipstrand) [inline README.md:203]: Nice that you added an example here!

File: README.md (hunk 3 above)
Comment (filipstrand) [reply README.md:219]: Yes for sure!

File: src/mflux/post_processing/generated_image.py (hunk 10 above)
Comment (filipstrand) [reply src/mflux/post_processing/generated_image.py:62]: Yes, only the `dev` model supports guidance. Right now, we pass the guidance value down to the lower level classes (regardless of model choice), but at some point deeper down, we do a check like the following: 

```
self.guidance_embedder = GuidanceEmbedder() if model_config == ModelConfig.FLUX1_DEV else None
```

and the guidance value is subsequently only used in case of "dev"

File: src/mflux/post_processing/generated_image.py (hunk 7 above)
Comment (filipstrand) [reply src/mflux/post_processing/generated_image.py:60]: Honestly don't remember why I put the explicit "None" strings here, but proper json nulls sounds better so lets go with your change here :)

File: tests/test_cli_argparser.py [hunk 16]
```diff
@@ -0,0 +1,258 @@
+import json
+from pathlib import Path
+from unittest.mock import patch
+
+import pytest
+
+from mflux.ui.cli.parsers import CommandLineParser
```
Comment (filipstrand) [inline tests/test_cli_argparser.py:7]: Really nice that you added these tests!

File: src/mflux/config/runtime_config.py [hunk 17]
```diff
@@ -39,6 +39,14 @@ def precision(self) -> mx.Dtype:
     def num_train_steps(self) -> int:
         return self.model_config.num_train_steps
 
+    @property
+    def init_image_path(self) -> int:
```
Comment (filipstrand) [inline src/mflux/config/runtime_config.py:43]: return types is wrong here

File: src/mflux/config/runtime_config.py [hunk 18]
```diff
@@ -39,6 +39,14 @@ def precision(self) -> mx.Dtype:
...
+        return self.config.init_image_path
+
+    @property
+    def init_image_strength(self) -> int:
```
Comment (filipstrand) [inline src/mflux/config/runtime_config.py:47]: return type is wrong here

File: README.md (hunk 2 above)
Comment (filipstrand) [reply README.md:198]: Great to also added the schema here! Make sense - lets start start with this and update the schema after that refactoring

File: README.md [hunk 19]
```diff
@@ -152,6 +152,76 @@ mflux-generate --model dev --prompt "Luxury food photograph" --steps 25 --seed 2
...
 
+- **`--config-from-metadata`** or **`-C`** (optional, `str`): [EXPERIMENTAL] Path to a prior file saved via `--metadata`, or a compatible handcrafted config file adhering to the expected args schema.
+
+<details>
+<summary>parameters supported by config files</summary>
+
+#### How configs are used
+
+- all config properties are optional and applied to the image generation if applicable
+- invalid or incompatible properties will be ignored
```
Comment (filipstrand) [inline README.md:163]: An alternative to this would be that we crash or something if the config file does not match the schema. Or ignoring incompatible properties as we do now, but maybe write out some kind of warning on what was ignored? 

A scenario that could potentially happen is that one of the properties is slightly misspelled and would be silently ignored (especially if the metadata file is constructed manually), say 

`"quantise": 4` instead of `"quantize": 4,` 

But this scenario would probably not be that common since I imagine that users would mostly edit existing config files.

**Comment by filipstrand** [comment]
@anthonywu I think this PR is looking really good and this is a nice feature that takes advantage of the metadata files - I initially just added them in case it was tricky for people to read the metadata embedded in the generated images themselves, and I was thinking that maybe it was a bit of a redundant feature to have, but now it feels like they have a proper purpose! 

Just had some smaller comments on this PR, if you mind having a look again.  After that, I think we can merge this and do a `0.4.0 release` (I'll just add a img2img example in the readme after this one is merged). 

Btw, I sent you an invite as a formal collaborator. Really appreciate your continuous involvement with this project!!

File: README.md (hunk 2 above)
Comment (anthonywu) [reply README.md:198]: todo:

- [ ] add support for `--config-from-metadata` in controlnet workflow

File: src/mflux/config/runtime_config.py (hunk 5 above)
Comment (anthonywu) [reply src/mflux/config/runtime_config.py:49]: todo:

- [ ] refactor config classes to have clarity on separation of responsibility and scope

File: src/mflux/post_processing/generated_image.py (hunk 10 above)
Comment (anthonywu) [reply src/mflux/post_processing/generated_image.py:62]: blocker/todo for #84

- [ ] guidance should be turned on for dev-like models as well (e.g. Freepik in #84)

File: src/mflux/ui/cli/parsers.py [hunk 20]
```diff
@@ -42,14 +56,74 @@ def add_output_arguments(self) -> None:
...
     def add_controlnet_arguments(self) -> None:
-        self.add_argument("--controlnet-image-path", type=str, required=True, help="Local path of the image to use as input for controlnet.")
+        self.supports_controlnet = True
+        self.add_argument("--controlnet-image-path", type=str, required=False, help="Local path of the image to use as input for controlnet.")
         self.add_argument("--controlnet-strength", type=float, default=ui_defaults.CONTROLNET_STRENGTH, help=f"Controls how strongly the control image influences the output image. A value of 0.0 means no influence. (Default is {ui_defaults.CONTROLNET_STRENGTH})")
         self.add_argument("--controlnet-save-canny", action="store_true", help="If set, save the Canny edge detection reference input image.")
 
+    def add_metadata_config(self) -> None:
+        self.supports_metadata_config = True
+        self.add_argument("--config-from-metadata", "-C", type=Path, required=False, default=argparse.SUPPRESS, help="Re-use the parameters from prior metadata. Params from metadata are secondary to other args you provide.")
```
[the remark this answers is not in the document]
Comment (anthonywu) [reply src/mflux/ui/cli/parsers.py:72]: @filipstrand what do you think about simplifying the flag to `--config` and keep `-C` or maybe generically `--config-from-file` / `-F`

I think both have many precedents in CLI tools.

File: src/mflux/config/runtime_config.py (hunk 18 above)
Comment (anthonywu) [reply src/mflux/config/runtime_config.py:47]: will fix in next commit

File: README.md (hunk 19 above)
Comment (anthonywu) [reply README.md:163]: proposing a standalone change to:

- [ ] introduce a `--strict` mode global flag such that at every step of the pipeline, missed expectations terminate the program. Examples:
  - [ ] dimensions are not multiples of 16
  - [ ] lora scale > 1.0, we don't want to `ceil` it for you
  - [ ] init image strength would cause no influence, let the user redo
  - [ ] guidance is passed in for non-accepting models (schnell)
  - [ ] there is a misspelled config, don't ignore it

and conversely, introduce a warning-printing workflow such that all of the above prints out colorful messages but will continue execution.

File: README.md (hunk 14 above)
Comment (anthonywu) [reply README.md:207]: I was conflicted whether I should have left model as the last required positional argument. If you're comfortable, I can make it optional and we just have to add additional catching logic when people don't identify it in the config files. The outcome will be that no argument is required!

Comment (anthonywu) [reply README.md:207]: updated in latest commit to not require `--model`

File: src/mflux/config/runtime_config.py (hunk 17 above)
Comment (anthonywu) [reply src/mflux/config/runtime_config.py]: fixed in 960447f92e768abaa49ffdcb5730fef535dc2fd8

File: README.md (hunk 14 above)
Comment (filipstrand) [reply README.md:207]: Thanks!

File: README.md (hunk 19 above)
Comment (filipstrand) [reply README.md:163]: Sounds great!
